switch/switch.js

```diff
@@ -0,0 +1,142 @@
+import {getStyle, getPseudoStyle} from './style-utils.js'
+
+class Switch extends HTMLInputElement
+{
+    #isDragging = false;
+    #recentlyDragged = false;
+    #thumbsize = 0;
+    #padding = 0;
+    #bounds = {
+        lower: 0,
+        middle: 0,
+        upper: 0,
+    };
+  
+    connectedCallback()
+    {
+        // There is some racing condition, no clue as to what, this delay seems to circumvent it for now
+          const thumbsize = getPseudoStyle(this, 'width');
+          const padding = getStyle(this, 'padding-left') + getStyle(this, 'padding-right');
+
+          this._thumbsize = thumbsize;
```

Comment:
```suggestion
          this.#thumbsize = thumbsize;
```

need to update more of these! i dont think this github ui is great for this particular edit 🙂